### DISCUSSION TITLE
Multi-value variable with wildcard not correctly expanded with ${var:lucene} for Elastic Search

### DISCUSSION THREAD

**Discussion by cadegenn** [grafana/grafana#58402] [Data Sources]
Please include this information:
### What Grafana version are you using?
v5.3.1

### What datasource are you using?
Elastic Search

### What OS are you running grafana on?
CentOS 7

### What did you do?
* create a variable type custom containing wildcards :
hostPrefix = prefixA*,prefixB*,prefixC*
* create a graph with query
my_host: ${hostPrefix:lucene}

### What was the expected result?
If we select multiple values, the expanded query should be
`my_host: (prefixA* OR prefixB* OR prefixC*)`

### What happened instead?
Instead, the query is
`my_host: ("prefixA\*" OR "prefixB\*" OR "prefixC\*")`

### raw json from query inspector should be
`"query":"my_host: (prefixA*" OR prefixB*)"`

### raw json from query inspector actually is
```

xhrStatus:"complete"
request:Object
method:"POST"
url:"api/datasources/proxy/8/_msearch"
data:"{"search_type":"query_then_fetch","ignore_unavailable":true,"index":["logstash-2018.10.18","logstash-2018.10.19"]} {"size":0,"query":{"bool":{"filter":[{"range":{"@timestamp":{"gte":"1539900000000","lte":"1539953898272","format":"epoch_millis"}}},{"query_string":{"analyze_wildcard":true,"query":"my_host: (\"prefixA\\*\" OR \"prefixB\\*\")"}}]}},"aggs":{"2":{"date_histogram":{"interval":"1m","field":"@timestamp","min_doc_count":0,"extended_bounds":{"min":"1539900000000","max":"1539953898272"},"format":"epoch_millis"},"aggs":{}}}} "


```
the interresting part is :
`"query":"my_host: (\"prefixA\\*\" OR \"prefixB\\*\")"`

Thank you for the great job.

**Comment by marefr** [comment]
Related to #10310

**Comment by torkelo** [comment]
This is not a bug, multi valued values with lucene strings is not supported, this is a feature request

**Comment by Elfo404** [comment]
variables currently match against their literal value, this also happens for non-multi valued variables.

https://github.com/grafana/grafana/blob/2a3aa951631cc0df141e96b251bd6011746f2ff5/public/app/features/templating/formatRegistry.ts#L245

note that changing this behaviour would probably be a breaking change.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
